joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/dataflow/DataflowTest.scala

```diff
@@ -695,4 +695,16 @@ class DataflowTest extends DataFlowCodeToCpgSuite {
       sink.reachableBy(identifierSource).size shouldBe 1
     }
   }
+  "Field access on TemplatedDom directly" should {
+    val cpg = code("""
+                     |import { useRouter } from "next/router";
+                     |
+                     |const tabComponentType = (<Tab title={"typeComponent"} />).type;
+                     |
+        |""".stripMargin)
+    "Not throw error and get handled it gracefully" in {
+      val List(x) = cpg.identifier("tabComponentType").l
```

Comment:
Perhaps just generate the CPG in here, then you could remove the query, since the error happens during a pass in any case.

Since you're only testing that an unhandled expression doesn't throw an error, you could do something like:

```scala
noException should be thrownBy code(""" ... """)
```

Reply:
For some reason the entire process of CPG generation doesn't happen in the `code` method invocation. You need to run some query to have CPG generated along with overlay passed getting executed.

Just now I tried what you suggested, and it doesn't throw the error and fail the test case when I try reverting the fix.

Reply:
Okay I see, must be that there is some "lazy" element implicit in the CPG test generation